index.html

```diff
@@ -3,47 +3,244 @@
+          } else if (options.filtered === 'midwest') {
+            data = data.filter(function (d) { return d.region === "Midwest"})
+          } else if (options.filtered === 'west') {
+            data = data.filter(function (d) { return d.region === "West"})
+     	}
+    }
+
+    var table = document.getElementById('data-table');
+    var tbody = table.getElementsByTagName('tbody')[0];
+
+    // Remove any existing rows
+    tbody.innerHTML = '';
+
+    // Add a new row for each datum (item in the 'data' array)
+    data.forEach(function (d) {
+      var row = document.createElement('tr');
+
+      var state = document.createElement('td');
+      state.innerHTML = d.state;
+      row.appendChild(state);  
+
+      var employment = document.createElement('td');
+      employment.innerHTML = d.employment;
+      row.appendChild(employment);
+
+      var region = document.createElement('td');
+      region.innerHTML = d.region;
+      row.appendChild(region);
+
+      var barWrapper = document.createElement('td');
+      var bar = document.createElement('div');
+      bar.setAttribute('class', 'bar');
+
+      var employmentPercentage = d.employment / MAX_EMPLOYMENT;
+      bar.setAttribute('style', 'width: ' + (employmentPercentage * 100) + '%;');
+
+      barWrapper.appendChild(bar);
+      row.appendChild(barWrapper);
+
+      tbody.appendChild(row);
+    });
+  }
+
+  updateTable();
+
+  document.getElementById('sort-asc').addEventListener('click', function () {
+    if (options.sorted === 'asc') {
+      options.sorted = false;
+    } else {
+      options.sorted = 'asc';
+    }
+
+    updateTable();
+  });
+
+  document.getElementById('sort-desc').addEventListener('click', function () {
+    if (options.sorted === 'desc') {
+      options.sorted = false;
+    } else {
+      options.sorted = 'desc';
+    }
+
+    updateTable();
+  });
+
+  document.getElementById('filter-ne').addEventListener('click', function () {
+    if (options.filtered === 'northeast') {
+      options.filtered = false;
+    } else {
+      options.filtered = 'northeast';
+    }
+    updateTable();
+  });
+
+  document.getElementById('filter-so').addEventListener('click', function () {
+    if (options.filtered === 'south') {
+      options.filtered = false;
+    } else {
+      options.filtered = 'south';
+    }
+    updateTable();
+  });
+
+  document.getElementById('filter-mw').addEventListener('click', function () {
+    if (options.filtered === 'midwest') {
+      options.filtered = false;
+    } else {
+      options.filtered = 'midwest';
+    }
+    updateTable();
+  });
+
+  document.getElementById('filter-we').addEventListener('click', function () {
+    if (options.filtered === 'west') {
+      options.filtered = false;
+    } else {
+      options.filtered = 'west';
+    }
+    updateTable();
+  });
+
+
+</script>
+
+  <p>Source: <a href="http://www.bls.gov/home.htm">Bureau of Labor Statistics</a></p>
```

Comment:
This works fine, but it's a best practice to have the scripts be the last thing on the page. This is because when the browser will draw the page in order, stopping to process each thing that it encounters. Here, our script is small and inline, so this isn't a big issue, but soon we'll be linking to a script in an external file. The result would be that this <p> would not be shown on the screen until after the script is fully downloaded.